.github/workflows/GHA-Spotbugs.yml

```diff
@@ -74,4 +74,91 @@ jobs:
         with:
           name: newrelic-api-spotbugs-results
           path: 'newrelic-api/build/spotbugs/main.*'
-          retention-days: 5
\ No newline at end of file
+          retention-days: 5
+
+  publish:
+    needs: spotbugs
+    runs-on: ubuntu-20.04
+    concurrency: gh-pages
+    steps:
+      - name: Set branch name env variable
+        env:
+          default_branch: 'main'
+        run: |
+          echo "BRANCH_NAME=${{ inputs.agent-ref ||  env.default_branch }}" >> $GITHUB_ENV
+
+      - name: Checkout GhPages branch
+        uses: actions/checkout@v3
+        with:
+          ref: 'gh-pages'
+          path: 'gh-pages'
+
+      - name: Configure Git
+        run: |
+          git config --global user.name "GitHub Actions Bot"
+          git config --global user.email "actions@users.noreply.github.com"
+
+      - name: Clear any artifacts from previous runs
+        run: |
+          rm -rf gh-pages/spotbugs/${{ env.BRANCH_NAME }}
+
+      # Download reports
+      - name: Download spotbugs reports 1
+        uses: actions/download-artifact@v3
+        with:
+          name: newrelic-agent-spotbugs-results
+          path: gh-pages/spotbugs/${{ env.BRANCH_NAME }}/newrelic-agent-spotbugs-results
+
+      - name: Download spotbugs reports 2
+        uses: actions/download-artifact@v3
+        with:
+          name: newrelic-weaver-spotbugs-results
+          path: gh-pages/spotbugs/${{ env.BRANCH_NAME }}/newrelic-weaver-spotbugs-results
+
+      - name: Download spotbugs reports 3
+        uses: actions/download-artifact@v3
+        with:
+          name: agent-bridge-spotbugs-results
+          path: gh-pages/spotbugs/${{ env.BRANCH_NAME }}/agent-bridge-spotbugs-results
+
+      - name: Download spotbugs reports 4
+        uses: actions/download-artifact@v3
+        with:
+          name: agent-model-spotbugs-results
+          path: gh-pages/spotbugs/${{ env.BRANCH_NAME }}/agent-model-spotbugs-results
+
+      - name: Download spotbugs reports 5
+        uses: actions/download-artifact@v3
+        with:
+          name: newrelic-api-spotbugs-results
+          path: gh-pages/spotbugs/${{ env.BRANCH_NAME }}/newrelic-api-spotbugs-results
+
+      # By default, GH-Pages only recognizes index.html files as landing pages.
```

Comment:
For a future task, we can modify the name of the html file generated by spotbugs so we can eliminate this step.